store/inputValidation.go

```diff
@@ -0,0 +1,86 @@
+package store
+
+import (
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/xmidt-org/argus/model"
+)
+
+var (
+	idFormatRegex     *regexp.Regexp
+	bucketFormatRegex *regexp.Regexp
+)
+
+var (
+	errInvalidID               = BadRequestErr{Message: "Invalid ID format. Expecting the format of a SHA-256 message digest."}
+	errIDMismatch              = BadRequestErr{Message: "IDs must match between the URL and payload."}
+	errDataFieldMissing        = BadRequestErr{Message: "Data field must be set in payload."}
+	errBodyReadFailure         = BadRequestErr{Message: "Failed to read body."}
+	errPayloadUnmarshalFailure = BadRequestErr{Message: "Failed to unmarshal json payload."}
+	errInvalidBucket           = BadRequestErr{Message: "Invalid bucket format."}
+)
+
+func init() {
+	idFormatRegex = regexp.MustCompile(`^[0-9a-f]{64}$`)
+	bucketFormatRegex = regexp.MustCompile("^[0-9a-z][0-9a-z-]{1,61}[0-9a-z]$")
+}
```

Comment:
What is the benefit of doing this rather than setting these in their var block?

Reply:
Also - would it make more sense to make these configuration values just so if we want to change them in the future it's a config change rather than a new release?

Reply:
No benefit! Good catch! 👍 

I think so? We could create a new config section:
```
userInputValidation:
    itemMaxTTL: "24h"
    itemIDFormatRegex: "^[0-9a-f]{64}$"
    bucketFormatRegex: "^[0-9a-z][0-9a-z-]{1,61}[0-9a-z]$"
```
It'll just be up to folks who configure this to make sure the format is compatible with the backend they use.

Reply:
Also, I would make the zero values of `userInputValidation` useful.  That has the side effect of making this configuration section optional, as the application will have defaults that work in most cases.  Configuration would simply override the defaults specified in a package-scoped `var` block.